### PR TITLE
[ie/Chaturbate] Support alternate domains

### LINKED ISSUE #10594: Add chaturbate support For other extentions
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Example URLs

https://chaturbate.global
https://chaturbate.eu

### Provide a description that is worded well enough to be understood

Well chaturbate has many streaming sites , like .com , .global , .eu .
From last 35 hours .com is down and other two are online .
And all three sites are similiar , all uses hls.
Just modifying regex for site will work , i guess.
If nobody is wiling to do it , i can try latter sometime.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU https://chaturbate.global/vanandjuani/
[debug] Command-line config: ['-vU', 'https://chaturbate.global/vanandjuani/']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.07.09 from yt-dlp/yt-dlp [7ead7332a]
[debug] Lazy loading extractors is disabled
[debug] Python 3.12.4 (CPython x86_64 64bit) - Linux-6.9.10-200.fc40.x86_64-x86_64-with-glibc2.39 (OpenSSL 3.2.1 30 Jan 2024, glibc 2.39)
[debug] exe versions: ffmpeg 6.1.1 (fdk,setts), ffprobe 6.1.1
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2023.05.07, mutagen-1.47.0, requests-2.31.0, sqlite3-3.45.1, urllib3-1.26.19, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, websockets
[debug] Loaded 1834 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/latest/download/_update_spec
Current version: stable@2024.07.09 from yt-dlp/yt-dlp
Latest version: stable@2024.07.25 from yt-dlp/yt-dlp
ERROR: You installed yt-dlp from a manual build or with a package manager; Use that to update
[generic] Extracting URL: https://chaturbate.global/vanandjuani/
[generic] vanandjuani: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] vanandjuani: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://chaturbate.global/vanandjuani/
Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/yt_dlp/YoutubeDL.py", line 1622, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/yt_dlp/YoutubeDL.py", line 1757, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 740, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/yt_dlp/extractor/generic.py", line 2521, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://chaturbate.global/vanandjuani/
```

### PR DESCRIPTION
**Pull request by manavchaudhary1**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

ADD DESCRIPTION HERE

Fixes #10594 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [x] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/chaturbate.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 class ChaturbateIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:[^/]+\.)?chaturbate\.com/(?:fullvideo/?\?.*?\bb=)?(?P<id>[^/?&#]+)'
+    _VALID_URL = r'https?://(?:[^/]+\.)?chaturbate\.(?P<domain>com|eu|global)/(?:fullvideo/?\?.*?\bb=)?(?P<id>[^/?&#]+)'
```
Comment (bashonly) [inline yt_dlp/extractor/chaturbate.py:12]:
```suggestion
    _VALID_URL = r'https?://(?:[^/]+\.)?chaturbate\.(?P<tld>com|eu|global)/(?:fullvideo/?\?.*?\bb=)?(?P<id>[^/?&#]+)'
```

File: yt_dlp/extractor/chaturbate.py [hunk 2]
```diff
@@ -34,10 +34,12 @@ class ChaturbateIE(InfoExtractor):
     _ROOM_OFFLINE = 'Room is currently offline'
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        mobj = self._match_valid_url(url)
+        video_id = mobj.group('id')
+        domain = mobj.group('domain')
```
Comment (bashonly) [inline yt_dlp/extractor/chaturbate.py:39]:
```suggestion
        video_id, tld = self._match_valid_url(url).group('id', 'tld')
```

File: yt_dlp/extractor/chaturbate.py [hunk 3]
```diff
@@ -34,10 +34,12 @@ class ChaturbateIE(InfoExtractor):
...
 
         webpage = self._download_webpage(
-            f'https://chaturbate.com/{video_id}/', video_id,
+            f'https://chaturbate.{domain}/{video_id}/', video_id,
```
Comment (bashonly) [inline yt_dlp/extractor/chaturbate.py:42]:
```suggestion
            f'https://chaturbate.{tld}/{video_id}/', video_id,
```

**Review by bashonly — CHANGES_REQUESTED** [review]
add tests for the newly supported domains please (they can be `only_matching`)

**Comment by manavchaudhary1** [comment]
Ok i'll add tests

**Comment by manavchaudhary1** [comment]
@bashonly should i rewrite the tests ?

**Review by bashonly — APPROVED** [review]

**Pull request merged**